signatures/pdf_sign_new_page.go

```diff
@@ -0,0 +1,215 @@
+/*
+ * This example showcases how to append a new page with signature to a PDF document.
+ * The file is signed using a private/public key pair.
+ *
+ * $ ./pdf_sign_new_page <INPUT_PDF_PATH> <OUTPUT_PDF_PATH>
+ */
+package main
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/unidoc/unipdf/v3/annotator"
+	"github.com/unidoc/unipdf/v3/common/license"
+	"github.com/unidoc/unipdf/v3/core"
+	"github.com/unidoc/unipdf/v3/model"
+	"github.com/unidoc/unipdf/v3/model/sighandler"
+)
+
+const licenseKey = `
+-----BEGIN UNIDOC LICENSE KEY-----
+Free trial license keys are available at: https://unidoc.io/
+-----END UNIDOC LICENSE KEY-----
+`
+
+func init() {
+	// Enable debug-level logging.
+	// unicommon.SetLogger(unicommon.NewConsoleLogger(unicommon.LogLevelDebug))
+
+	err := license.SetLicenseKey(licenseKey, `Company Name`)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func main() {
+	args := os.Args
+	if len(args) < 3 {
+		log.Fatalln("Usage: go run pdf_sign_new_page INPUT_PDF_PATH OUTPUT_PDF_PATH")
+	}
+
+	inputPath := args[1]
+	outputPath := args[2]
+
+	// Read the original file.
+	f, err := os.Open(inputPath)
+	if err != nil {
+		log.Fatalf("Fail: %v\n", err)
+	}
+	defer f.Close()
+
+	pdfReader, err := model.NewPdfReader(f)
+	if err != nil {
+		log.Fatalf("Fail: %v\n", err)
+	}
+
+	// Add new page and write it into a temporary file.
+	tempPath, err := addPage(pdfReader)
```

Comment:
You should write to a memory buffer instead of a temporary file. Something like:
```go
	// Add new page and write to a memory buffer.
	buf, err := addPage(pdfReader)
	if err != nil {
		log.Fatalf("Fail: %v\n", err)
	}

	pdfReader, err = model.NewPdfReader(bytes.NewReader(buf.Bytes()))
	if err != nil {
		log.Fatalf("Fail: %v\n", err)
	}

	totalPage, err := pdfReader.GetNumPages()
	if err != nil {
		log.Fatalf("Fail: %v\n", err)
	}

	// Add signature and write it to output.pdf file.
	err = addSignature(pdfReader, totalPage, outputPath)
	if err != nil {
		log.Fatalf("Fail: %v\n", err)
	}

	fmt.Println("Done")
}

func addPage(reader *model.PdfReader) (*bytes.Buffer, error) {
	writer, err := reader.ToWriter(&model.ReaderToWriterOpts{})
	if err != nil {
		return nil, err
	}

	if err = writer.AddPage(model.NewPdfPage()); err != nil {
		return nil, err
	}

	buf := bytes.NewBuffer(nil)
	if err = writer.Write(buf); err != nil {
		return nil, err
	}

	return buf, nil
}
```